### PR TITLE
feat: LEAP-58: Proxy URL for Taxonomy if possible

### PR DESCRIPTION
**Pull request by hlomzik**
add presignUrlForProject to AppStore, and invoke it to rewrite Taxonomy apiUrl with presigned version when appropriate.

### CODE REVIEW

File: src/stores/AppStore.js [hunk 1]
```diff
@@ -622,6 +625,18 @@ export default types
       }, 'Error during reject, try again');
     }
 
+    /**
+     * Exchange storage url for presigned url for task
+     */
+    async function presignUrlForProject(url) {
+      // Event invocation returns array of results for all handlers.
+      const urls = await getEnv(self).events.invoke('presignUrlForProject', self, url);
```
Comment (hlomzik) [inline src/stores/AppStore.js:633]: let's use this `events` getter right away
```suggestion
      const urls = await self.events.invoke('presignUrlForProject', self, url);
```

File: src/tags/control/Taxonomy/Taxonomy.js [hunk 2]
```diff
@@ -518,6 +518,7 @@ const Model = types
         if (!self.isLoadedByApi) return Super.updateValue?.(store);
 
         self._api = parseValue(self.apiurl, store.task.dataObj);
+        self._api = (yield store.presignUrlForProject(self._api)) ?? self._api;
```
Comment (hlomzik) [inline src/tags/control/Taxonomy/Taxonomy.js:521]: maybe a comment?
```suggestion
        // trying to presign this url if needed and if handler is passed into LSF
        self._api = (yield store.presignUrlForProject(self._api)) ?? self._api;
```

**Review by bmartel — APPROVED** [review]

**Comment by jombooth** [comment]
ignoring e2e failure because e2e tests passed at [11fc4ce](https://github.com/HumanSignal/label-studio-frontend/pull/1592/commits/11fc4ce2c496e9af7fb3966b08c0b9ce35a56303) but not [56a167e](https://github.com/HumanSignal/label-studio-frontend/pull/1592/commits/56a167eb97f79e77236a55617c2bff133683edcf), though there are no code changes between these two commits.

**Pull request merged**